internal/cri/server/container_stats_list.go

```diff
@@ -141,12 +141,9 @@ func (c *criService) toContainerStats(
 			handler, err = c.getMetricsHandler(ctx, cntr.SandboxID)
 			if err != nil {
 				// If the sandbox is not found, it may have been removed. we need to check container whether it is still exist
```

Comment:
I disagree on the need here to check container exist.. While it may be extra info for the warnf I don't see a need to check then modify the warn or exit metrics processing .. there may be any number of reasons for there to be no valid metrics handler for a container's sandbox...  I can't think of a reason to fail all stats because one container / container sandbox is in a transition state or won't be supported by platform etc..  Just log warn and skip on the error to determine the container sandbox's metrics handler.

Reply:
Fixed to skip the container in case of any error in `getMetricsHandler`